libs/MVS/SceneDensify.cpp

```diff
@@ -1962,9 +1963,17 @@ void Scene::DenseReconstructionEstimate(void* pData)
 			}
 			// try to load already compute depth-map for this image
 			if (depthmapComputed && data.nFusionMode >= 0) {
+				const std::string depthMapName(ComposeDepthFilePath(data.scene.images[idx].ID, "dmap"));
+				cv::Size depthMapSize;
+				if (GetDepthMapHeaderSize(depthMapName, depthMapSize) && depthMapSize != data.scene.images[idx].GetSize()) {
+					depthData.Load(depthMapName);
+					DepthMap& depthMap = depthData.depthMap;
+					cv::resize(depthMap, depthMap, data.scene.images[idx].GetSize(), 0, 0, cv::INTER_NEAREST);
+					depthData.Save(depthMapName);
```

Comment:
it is ok if you use `depthData` to load the depth-map, but after saving it back to disk you should release it from `depthData` as well

and you can combine this with the next check that loads the depth-map, and if we that check is true we do not release it